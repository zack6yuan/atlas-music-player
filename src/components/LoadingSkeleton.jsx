export default function LoadingSkeleton() {
  return (
    <div role="status" className="max-w-sm animate-pulse">
      <div className="mb-4 h-2.5 w-20 rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <div className="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"></div>
      <span className="sr-only">Loading...</span>
    </div>
  );
}
