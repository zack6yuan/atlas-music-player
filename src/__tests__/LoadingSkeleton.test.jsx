import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import LoadingSkeleton from "../components/LoadingSkeleton"

test("Loading Skeleton renders correctly", () => {
    const { container } = render(
        <LoadingSkeleton />
    );
    expect(container).toMatchSnapshot();
});