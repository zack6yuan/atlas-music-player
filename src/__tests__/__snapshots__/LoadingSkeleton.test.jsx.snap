// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading Skeleton renders correctly 1`] = `
<div>
  <div
    class="max-w-sm animate-pulse"
    role="status"
  >
    <div
      class="mb-4 h-2.5 w-20 rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[193px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <div
      class="mb-2.5 h-2 max-w-[160px] rounded-full bg-gray-200 dark:bg-gray-700"
    />
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
